retrievalmarket/impl/client.go

```diff
@@ -72,16 +75,17 @@ func dispatcher(evt pubsub.Event, subscriberFn pubsub.SubscriberFn) error {
 var _ retrievalmarket.RetrievalClient = &Client{}
 
 // NewClient creates a new retrieval client
-func NewClient(network rmnet.RetrievalMarketNetwork, multiStore *multistore.MultiStore, dataTransfer datatransfer.Manager, node retrievalmarket.RetrievalClientNode, resolver discovery.PeerResolver, ds datastore.Batching) (retrievalmarket.RetrievalClient, error) {
+func NewClient(network rmnet.RetrievalMarketNetwork, carPath string, dataTransfer datatransfer.Manager, node retrievalmarket.RetrievalClientNode, resolver discovery.PeerResolver, ds datastore.Batching) (retrievalmarket.RetrievalClient, error) {
```

Comment:
The retrieval client puts files in the directory that carPath points to. It's just a little simpler for the caller if it specifies a directory to put stuff in rather than having to specify a new filename every time.

Reply:
I agree with Hannah, and I also pointed out in other comments that this library should not be actively managing files on the filesystem. I too agree with @dirkmc that it's a harder change to implement, but it shouldn't be too hard. It's just good behaviour to let the user specify where they want the output of the operation (CAR file) placed.

Reply:
Currently the carPath is initialized as a directory inside the DAG store:
```
carsPath := filepath.Join(r.Path(), DefaultDAGStoreDir, "retrieval-cars")
```

And the file path is derived from the deal ID:
```
carFilePath := filepath.Join(c.carPath, dealID.String())
```

This has the nice property that the name of the file containing the retrieval data is the same as the deal ID.

If we instead pass the file path as a parameter to the `Retrieve()` function we will not be able to name the file the same as the deal ID (the deal ID gets created inside the `Retrieve()` function.

Reply:
Discussed sync with @raulk 
Resolution is to adopt Hannah's suggestion

Reply:
follow-up -- upon further review of the Lotus code, should Retrieve just take a blockstore? This might make IPFS blockstore integration easier.

Reply:
It needs to be a file so that we can safely support resumption

Reply:
Fixed in https://github.com/filecoin-project/go-fil-markets/pull/576/commits/6b375f9a43cf6c233c5722310c296e330126a861